fix(AIP-132): Require documentation for ordering not matching field type (#1484)
googleapis/aip: aip/general/0132.md

  equivalent.
- Subfields are specified with a `.` character, such as `foo.bar` or
  `address.street`.

> - The resulting list order **should** usually be numeric for number-typed fields

Reviewer: ```suggestion
- The resulting list order **should** be numeric for number-typed fields
```
Let's drop the squishy language following a requirement word

>   and lexicographic for other types. However, APIs **may** choose to use a

Reviewer: I really like the succinct structure here. That said, I think we need to be a little more specific than "other types" - `google.protobuf.Timestamp` wouldn't use lexicographic. 

```
- The resulting list order **should** be based on the field type's natural comparator e.g. numerics ordered numerically, strings ordered lexicographically, etc. However, APIs **may** choose to use a different ordering;
if so, it **must** be documented in the `order_by` definition.
  - Furthermore, [well-known][] types, like `Timestamp` and `Duration` are compared as their representative
    type; `Timestamp` is compared as time e.g. before or after, `Duration` is compared numerically e.g. larger
    or smaller.     

[well-known]: https://protobuf.dev/reference/protobuf/google.protobuf
```

I might be over-writing at this point though. WDYT?

Author: I really like that phrasing!  IMO its best to be as explicit as possible.  I just committed it

  equivalent.
- Subfields are specified with a `.` character, such as `foo.bar` or
  `address.street`.
- The resulting list order **should** be based on the field type's natural
  comparator e.g. numerics ordered numerically, strings ordered
  lexicographically, etc. However, APIs **may** choose to use a different
  ordering; if so, it **must** be documented in the `order_by` definition.
  - Furthermore, [well-known][] types, like `Timestamp` and `Duration` are
    compared as their representative type; `Timestamp` is compared as time e.g.

>     before or after, `Duration` is compared numerically e.g. larger or smaller.

Reviewer: ```suggestion
    before or after, `Duration` is compared as a quantity e.g. more or less.
```
After rereading, I think `quantity` is the better term, as a "numerical value multiple of a unit of measurement" per the ol'google. As such I think the terms `more` and `less` align a bit better. (I know, nit picks)

Author: Hmm, I don't have a strong opinion on this but am happy to commit it if you think its better.
